packages/collection/src/collection.ts

```diff
@@ -669,52 +669,93 @@ export class Collection<K, V> extends Map<K, V> {
+	 * The intersection method returns a new collection containing the items where the key is present in both collections.
 	 *
 	 * @param other - The other Collection to filter against
+	 * @example
+	 * ```ts
+	 * const col1 = new Collection([['a', 1], ['b', 2]]);
+	 * const col2 = new Collection([['a', 1], ['c', 3]]);
+	 * const intersection = col1.intersection(col2);
+	 * console.log(col1.intersection(col2));
+	 * // => Collection { 'a' => 1 }
+	 * ```
 	 */
-	public intersect<T>(other: ReadonlyCollection<K, T>): Collection<K, T> {
-		const coll = new this.constructor[Symbol.species]<K, T>();
-		for (const [k, v] of other) {
-			if (this.has(k) && Object.is(v, this.get(k))) {
-				coll.set(k, v);
-			}
+	public intersection<T>(other: ReadonlyCollection<K, T>): Collection<K, T | V> {
+		const coll = new this.constructor[Symbol.species]<K, T | V>();
+
+		for (const [k, v] of this) {
+			if (other.has(k)) coll.set(k, v);
 		}
 
 		return coll;
 	}
 
 	/**
-	 * The subtract method returns a new structure containing items where the keys and values of the original structure are not present in the other.
+	 * The union method returns a new collection containing the items where the key is present in either of the collections.
+	 *
+	 * @param other - The other Collection to filter against
+	 * @example
+	 * ```ts
+	 * const col1 = new Collection([['a', 1], ['b', 2]]);
+	 * const col2 = new Collection([['a', 1], ['c', 3]]);
+	 * const union = col1.union(col2);
+	 * console.log(union);
+	 * // => Collection { 'a' => 1, 'b' => 2, 'c' => 3 }
+	 * ```
+	 */
+	public union<T>(other: ReadonlyCollection<K, T>): Collection<K, T | V> {
+		return new this.constructor[Symbol.species]<K, T | V>([...this, ...other]);
+	}
+
+	/**
+	 * The difference method returns a new collection containing the items where the key is present in this collection and not the other.
 	 *
 	 * @param other - The other Collection to filter against
+	 * @example
+	 * ```ts
+	 * const col1 = new Collection([['a', 1], ['b', 2]]);
+	 * const col2 = new Collection([['a', 1], ['c', 3]]);
+	 * console.log(col1.difference(col2));
+	 * // => Collection { 'b' => 2 }
+	 * console.log(col2.difference(col1));
+	 * // => Collection { 'c' => 3 }
+	 * ```
 	 */
-	public subtract<T>(other: ReadonlyCollection<K, T>): Collection<K, V> {
+	public difference<T>(other: ReadonlyCollection<K, T>): Collection<K, V> {
 		const coll = new this.constructor[Symbol.species]<K, V>();
+
 		for (const [k, v] of this) {
-			if (!other.has(k) || !Object.is(v, other.get(k))) {
-				coll.set(k, v);
-			}
+			if (!other.has(k)) coll.set(k, v);
 		}
 
 		return coll;
 	}
 
 	/**
-	 * The difference method returns a new structure containing items where the key is present in one of the original structures but not the other.
+	 * The difference method returns a new collection containing the items where the key is present in this collection and not the other, and vice-versa.
```

Comment:
This can be much to unpack for the average user, while the terminology is correct, it may be too technical.

We can draw some inspiration about the wording from other languages, e.g. .NET's [`HashSet<T>.SymmetricExceptWith`](https://learn.microsoft.com/en-us/dotnet/api/system.collections.generic.hashset-1.symmetricexceptwith?view=net-7.0):

```suggestion
	 * Returns a new collection containing only the items where the keys are present in either collection, but not both.
```

The description above fits better with the mathematical description too: https://en.wikipedia.org/wiki/Symmetric_difference